Foreach swift (#5150)
Codecademy/docs: content/swift/concepts/dictionaries/terms/forEach/foreach.md

> ---
> Title: 'forEach'
> Description: 'The forEach method allows iteration over each key-value pair in a Swift dictionary.'
> Subjects:
>   - 'Mobile Development'
>   - 'Computer Science'
> Tags:
>   - 'Dictionary'
>   - 'Properties'
> CatalogContent:
>   - 'learn-swift'
>   - 'paths/build-ios-apps-with-swiftui'
> ---

Reviewer: ```suggestion
---
Title: '.forEach()'
Description: 'Iterates over each key-value pair in a Swift dictionary, applying a closure to both the key and value.'
Subjects:
  - 'Computer Science'
  - 'Mobile Development'
Tags:
  - 'Dictionary'
  - 'Properties'
CatalogContent:
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
---
```

Reviewer: Modifications:
1. description should always start with a verb
2. tags and subjects should be in alphabetical manner

> The **`forEach`** method is used to iterate over each key-value pair in a Swift dictionary.
> It applies a specified closure to each element in the dictionary.
> This method is particularly useful for performing actions or computations on each element
> without needing to manually handle key access.

Reviewer: ```suggestion
The **`.forEach()`** method is used to iterate over each key-value pair in a Swift dictionary. It applies a specified closure to each key-value pair, allowing for actions or computations to be performed on each element without manual key access. This method is particularly useful for executing side effects or processing elements concisely and functionally.
```

Reviewer: Written in a single line, made it more readable

## Syntax

```pseudo
dictionary.forEach { (key, value) in 
    // Code to execute for each key-value pair
}
```

> - dictionary:  This is the Swift dictionary you want to iterate over. It contains key-value pairs that you want to process.
> - foreach: This is the method called on the dictionary object. It takes a closure as its argument.
> - { (key, value) in: This is the closure syntax used with forEach. Think of it as a placeholder for each key-value pair in the dictionary.

Reviewer: ```suggestion
- `dictionary`: The variable name for the dictionary instance.
- `{ (key, value) in ... }`: Defines a closure where key and value are parameters representing each key-value pair from the `dictionary`. 
```

## Example

> Here's an example of using 'forEach' with a dictionary:
> ```swift
> let fruits = ["apple": 3, "banana": 2, "cherry": 5]
> 
> fruits.forEach { (key, value) in
>     print("There are \(value) \(key)s.")
> }
> ```
> if you run this code, this will be the output:
> ```shell
> There are 3 apples.
> There are 2 bananas.
> There are 5 cherrys.
> ```

Reviewer: ```suggestion

Here's an example of using `.forEach()` with a dictionary:

```swift
let fruits = ["apple": 3, "banana": 2, "cherry": 5]

fruits.forEach { (key, value) in
    print("There are \(value) \(key)s.")
}
```

The code above produces the following output:

```shell
There are 3 apples.
There are 2 bananas.
There are 5 cherrys.
```
```

Reviewer: Avoid any "you" "we" in the entry

Reviewer: At the end you can add a note saying that the order is not the same everytime. Example:

> **Note:** The order of key-value pairs iterated over in a Swift dictionary using forEach is not guaranteed and may vary between executions.

---
Title: 'forEach'
Description: 'The forEach method allows iteration over each key-value pair in a Swift dictionary.'
Subjects:

>   - 'iOS'

Author: but I noticed that iOS is already listed in the subjects.md file

Tags:
  - 'ForEach'
  - 'Dictionary'
  - 'Loops'
  - 'Collections'

>   - 'iOS'

Reviewer 2: ```suggestion
  - 'iOS'
```
The `iOS` named tag is not present in the tags.md list. You can add it if you want it to be included in the file.

Author: It is included in tags.md. I double-checked just to be sure, and it's there.
